pkg/controllermanager/controller/shoot/shoot_reference_control.go

```diff
@@ -252,6 +314,55 @@ func (r *shootReferenceReconciler) getUnreferencedSecrets(c client.Client, shoot
 	return secretsToRelease, nil
 }
 
+func (r *shootReferenceReconciler) getUnreferencedConfigMaps(c client.Client, shoot *gardencorev1beta1.Shoot) ([]corev1.ConfigMap, error) {
+	namespace := shoot.Namespace
+
+	configMaps := &corev1.ConfigMapList{}
+	if err := r.configMapLister(r.ctx, configMaps, client.InNamespace(namespace)); err != nil {
+		return nil, err
+	}
+
+	// Exit early if there are no ConfigMaps at all in the namespace
+	if len(configMaps.Items) == 0 {
+		return nil, nil
+	}
+
+	shoots := &gardencorev1beta1.ShootList{}
+	if err := c.List(r.ctx, shoots, client.InNamespace(namespace)); err != nil {
+		return nil, err
+	}
+
+	referencedConfigMaps := sets.NewString()
+	for _, s := range shoots.Items {
+		// Ignore own references if shoot is in deletion and references are not needed any more by Gardener.
+		if s.Name == shoot.Name && shoot.DeletionTimestamp != nil && !controllerutils.HasFinalizer(&s, gardencorev1beta1.GardenerName) {
+			continue
+		}
+		apiServerConfig := s.Spec.Kubernetes.KubeAPIServer
+		if apiServerConfig != nil &&
+			apiServerConfig.AuditConfig != nil &&
+			apiServerConfig.AuditConfig.AuditPolicy != nil &&
+			apiServerConfig.AuditConfig.AuditPolicy.ConfigMapRef != nil {
+
+			configMapRef := apiServerConfig.AuditConfig.AuditPolicy.ConfigMapRef
```

Comment:
Can you extract this to a separate function since you can also re-use it in `handleReferencedConfigMap`?